### PR TITLE
Move from github.com/DataDog/zstd to github.com/klauspost/compress/zstd

### PR DESCRIPTION
**Pull request by jb-abbadie**
## Description

This allow us to have a pure Go package which is much easier to package
on different OS/Architectures

 

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Checklist

- [X] Linter passes correctly
- [X] Add tests which fail without the change (if possible)
- [X] All tests passing
- [] Extended the README / documentation, if necessary

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Further comments

I will do a PR to put `CGO_ENABLED=0` in all of the CI once this PR is merged

### CODE REVIEW

File: chdecompressor/decompressor.go [hunk 1]
```diff
@@ -83,7 +84,8 @@ func (r *Reader) readNextBlock() error {
 			return fmt.Errorf("cannot decompress lz4 block: %w", err)
 		}
 	case zstdType:
-		r.data, err = zstd.Decompress(r.data, block)
+		r.data = r.data[:0] // Wipe the slice but keep allocated memory
```
Comment (LEijsackersCS) [inline chdecompressor/decompressor.go:87]: Why is this needed? We allocate the slice above.

File: go.mod [hunk 2]
```diff
@@ -23,6 +23,7 @@ require (
 	github.com/dgryski/go-rendezvous v0.0.0-20200823014737-9f7001d12a5f // indirect
 	github.com/frankban/quicktest v1.7.2 // indirect
 	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/klauspost/compress v1.15.11 // indirect
```
Comment (LEijsackersCS) [inline go.mod]: Should we not also remove the DataDog dependency?

**Review by LEijsackersCS — COMMENTED** [review]
Thanks for your PR! I have two comments.

File: chdecompressor/decompressor.go (hunk 1 above)
Comment (jb-abbadie) [reply chdecompressor/decompressor.go:87]: Its because the [new library is doing](https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.DecodeAll) : 
> Output will be appended to dst

Compared to datadog which just replace the buffer

File: go.mod (hunk 2 above)
Comment (jb-abbadie) [reply go.mod]: fixed

**Review by LEijsackersCS — APPROVED** [review]
LGTM!
